.github/workflows/ci.yml

```diff
@@ -61,9 +61,14 @@ jobs:
           key:
             cache|${{ runner.os }}|${{ env.PY }}|${{ hashFiles('pyproject.toml') }}|${{
             hashFiles('poetry.lock') }}|${{ hashFiles('.pre-commit-config.yaml') }}
-      - run: python -m pip install poetry poetry-dynamic-versioning
+      - name: Install poetry
         shell: bash
-      - run: poetry install
+        run: |
+          python -m pip install poetry poetry-dynamic-versioning
+          echo "::set-env name=PATH::$HOME/.poetry/bin:$PATH"
+      - run: |
+          poetry run python -m pip install pip -U
+          poetry install
```

Comment:
Please don't modify these lines.

First of all, You don't need `echo "::set-env name=PATH::$HOME/.poetry/bin:$PATH"` because you're installing poetry with pip. That is only used when installing it like in https://python-poetry.org/docs/#osx-linux-bashonwindows-install-instructions

Second, it doesn't make sense to upgrade pip after you've used it.

Finally, these steps are tightly coupled with cache, so I beg you to not touch them unless strictly required to have faster builds.